android/src/main/java/com/swmansion/reanimated/layoutReanimation/AnimationsManager.java

```diff
@@ -376,6 +380,13 @@ public boolean isLayoutAnimationEnabled() {
 
   private boolean removeOrAnimateExitRecursive(View view, ViewGroup parent, boolean shouldRemove) {
     int tag = view.getId();
+    ViewManager viewManager = resolveViewManager(tag);
+
+    if (viewManager != null && viewManager.getName().equals("RNSScreenStack")) {
+      cancelAnimationsRecursive(view);
+      return false;
+    }
```

Comment:
It's possible it's not just the screen that's being removed, but the whole stack.
In that case, we have to cancel the animations in the screen as well, which we do here.